ksql-common/src/main/java/io/confluent/ksql/errors/LogMetricAndContinueExceptionHandler.java

```diff
@@ -0,0 +1,49 @@
+/**
+ * Copyright 2017 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+package io.confluent.ksql.errors;
+
+import io.confluent.ksql.metrics.MetricCollectors;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.streams.errors.DeserializationExceptionHandler;
+import org.apache.kafka.streams.processor.ProcessorContext;
+import org.apache.kafka.streams.processor.internals.StreamThread;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Map;
+
+public class LogMetricAndContinueExceptionHandler implements DeserializationExceptionHandler {
+  private static final Logger log = LoggerFactory.getLogger(StreamThread.class);
+
+  @Override
+  public DeserializationHandlerResponse handle(final ProcessorContext context,
+                                               final ConsumerRecord<byte[], byte[]> record,
+                                               final Exception exception) {
+
+    log.warn("Exception caught during Deserialization, " +
+                    "taskId: {}, topic: {}, partition: {}, offset: {}",
+            context.taskId(), record.topic(), record.partition(), record.offset(),
+            exception);
+
+    MetricCollectors.recordError(record.topic());
+    return DeserializationHandlerResponse.CONTINUE;
```

Comment:
Shouldn't this return value depend on how `FAIL_ON_DESERIALIZATION` is configured? Do we want to continue if `FAIL_ON_DESERIALIZATION` is false? Presumably the query is failed elsewhere in the latter case. If so, a comment here would help.

Reply:
Thanks Apuva - if we fail then all processing stops; this impl is identical to the streams one. This is the side-channel where the user has the option to provide their own handler - however, a bad data format instance doesn't kill the stream processor.